provider/aws/aws_test.go

```diff
@@ -1734,6 +1747,19 @@ func TestAWSCreateRecordsWithALIAS(t *testing.T) {
 	}
 }
 
+func TestAWSAdjustEndpointsWithNilEndpoint(t *testing.T) {
+	provider, _ := newAWSProvider(t, endpoint.NewDomainFilter([]string{"ext-dns-test-2.teapot.zalan.do."}), provider.NewZoneIDFilter([]string{}), provider.NewZoneTypeFilter(""), defaultEvaluateTargetHealth, false, nil)
+
+	records := []*endpoint.Endpoint{
+		nil,
+	}
+
+	adjusted, err := provider.AdjustEndpoints(records)
+	require.NoError(t, err)
+
+	validateEndpoints(t, provider, adjusted, records)
```

Comment:
There's no adjustment to be made for `nil` so `adjusted` should match `records`.